site/app/libraries/database/DatabaseUtils.php

```diff
@@ -16,7 +16,13 @@ public static function formatQuery(string $sql, ?array $params): string {
             if ($param instanceof \DateTime) {
                 $param = DateUtils::dateTimeToString($param);
             }
-            $sql = preg_replace('/\?/', is_numeric($param) ? $param : "'{$param}'", $sql, 1);
+            if (gettype($param) === "array") {
+                $param = json_encode($param);
```

Comment:
json_encode seems to be a decent option but when it gets substituted in it'll produce `([1, 2, ...])`. Then `[` and `]` are invalid SQL. So if we use json_encode we should make sure those braces get removed so it's valid SQL.

Reply:
I realized that but then thought about [PostgreSQL's JSON functions](https://www.postgresql.org/docs/9.5/functions-json.html). I was not sure if Doctrine accepts an array in some method and then uses it as an argument in a call to a JSON function in the resulting query, in which case it may be correct to keep square brackets and also single quotes e.g. `json_array_elements_text('["foo", "bar"]')`. Could this be a possible scenario?

Reply:
Yeah I think using `json_array_elements_text` is a good option. Just make sure it works for both strings and integers (in the sense that it's valid SQL and collects the right data). Then we won't have to worry about some json edge case.

Reply:
`json_array_elements_text` creates issues concerning valid syntax. As of now, any issue related to some JSON edge case hasn't been discovered, handling it may result in unneeded additional complexity.
Still, I have added some code that deals with some JSON functions, but as it may be redundant, I've commented it. It can be used later if the need may arise...